pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -62,6 +89,220 @@ public PinotTaskManager(PinotHelixTaskResourceManager helixTaskResourceManager,
+    removeAllTasksFromCronExpressions(tableWithType);
+    _tableTaskSchedulerUpdaterMap.remove(tableWithType);
+  }
+
+  private synchronized void removeAllTasksFromCronExpressions(String tableWithType) {
+    Set<JobKey> jobKeys;
+    try {
+      jobKeys = _scheduledExecutorService.getJobKeys(GroupMatcher.anyJobGroup());
+    } catch (SchedulerException e) {
+      LOGGER.error("Got exception when fetching all jobKeys", e);
+      return;
+    }
+    for (JobKey jobKey : jobKeys) {
+      if (jobKey.getName().equalsIgnoreCase(tableWithType)) {
+        try {
+          _scheduledExecutorService.deleteJob(jobKey);
+        } catch (SchedulerException e) {
+          LOGGER.error("Got exception when deleting the scheduled job - {}", jobKey, e);
+        }
+      }
+    }
+    _tableTaskTypeToCronExpressionMap.remove(tableWithType);
+  }
+
+  public synchronized void subscribeTableConfigChanges(String tableWithType) {
+    if (_tableTaskSchedulerUpdaterMap.containsKey(tableWithType)) {
+      return;
+    }
+    TableTaskSchedulerUpdater tableTaskSchedulerUpdater = new TableTaskSchedulerUpdater(tableWithType, this);
+    _pinotHelixResourceManager.getPropertyStore()
+        .subscribeDataChanges(getPropertyStorePathForTable(tableWithType), tableTaskSchedulerUpdater);
+    _tableTaskSchedulerUpdaterMap.put(tableWithType, tableTaskSchedulerUpdater);
+    LOGGER.info("Trying to update task schedule for table: {}", tableWithType);
+    try {
+      updateCronTaskScheduler(tableWithType);
+    } catch (Exception e) {
+      LOGGER.error("Got exception during updateCronTaskScheduler for {}", tableWithType, e);
+    }
+  }
+
+  public synchronized void updateCronTaskScheduler(String tableWithType) {
+    if (_scheduledExecutorService == null) {
+      return;
+    }
+    TableConfig tableConfig = _pinotHelixResourceManager.getTableConfig(tableWithType);
+    TableTaskConfig taskConfig = tableConfig.getTaskConfig();
+    if (taskConfig == null) {
+      LOGGER.info("taskConfig is null, trying to remove all the tasks for table {} if any", tableWithType);
+      removeAllTasksFromCronExpressions(tableWithType);
+      return;
+    }
+    Map<String, Map<String, String>> taskTypeConfigsMap = taskConfig.getTaskTypeConfigsMap();
+    if (taskTypeConfigsMap == null) {
+      LOGGER.info("taskTypeConfigsMap is null, trying to remove all the tasks for table {} if any", tableWithType);
+      removeAllTasksFromCronExpressions(tableWithType);
+      return;
+    }
+    Map<String, String> taskToCronExpressionMap = getTaskToCronExpressionMap(taskTypeConfigsMap);
+    LOGGER.info("Got taskToCronExpressionMap {} ", taskToCronExpressionMap);
+    updateCronTaskScheduler(tableWithType, taskToCronExpressionMap);
+  }
+
+  private void updateCronTaskScheduler(String tableWithType, Map<String, String> taskToCronExpressionMap) {
+    if (_scheduledExecutorService == null) {
+      return;
+    }
+    if (_tableTaskTypeToCronExpressionMap.containsKey(tableWithType)) {
```

Comment:
(nit) use `get(tableWithType)` and `null` check to save one map lookup. Same for other `containsKey()` followed by `get()`